### DISCUSSION TITLE
Questions about default time derivative

### DISCUSSION THREAD

**Discussion by Richard-happy** [idaholab/moose#23753] [Q&A General] [answered]
[thermomech_step01.txt](https://github.com/idaholab/moose/files/10986150/thermomech_step01.txt)

For the input file (modules/combined/tutorials/introduction/thermal_mechanical/thermomech_step01.i),the default type about the first derivative of time is ImplicitEuler. What's type about the second derivative of time   ?

**Comment by GiudGiud** [comment]
Second derivative of time I m not sure.
I could track down an implementation for the CentralDifference time integrator and the NewmarkBeta time integrator.

@recuero @cticenhour

**Comment by cticenhour** [comment]
@GiudGiud is correct - both the [CentralDifference](https://mooseframework.inl.gov/source/timeintegrators/CentralDifference.html) and [NewmarkBeta](https://mooseframework.inl.gov/source/timeintegrators/NewmarkBeta.html) time integrators contain a second derivative calculation. While I don't see any guided examples that use these, the "Input Files" section on both of those pages contain test input files that might be used as an example. The Contact, Tensor Mechanics, FSI, and Electromagnetics modules all use this in some of their inputs.

**Reply by Richard-happy** [reply]
Thank you

**Comment by GiudGiud** [comment, marked as answer]
So I played with this a little. If you try to retrieve a 2nd order time derivative with a time integrator object that does not implement them (currently, someone could come and implement them for every time integrator), then you get an error. 

This PR can help you look at this
https://github.com/idaholab/moose/pull/23771

**Reply by Richard-happy** [reply]
Thank you
